harvard-lts/CURIOSity
# Updating show pages to use mps-embed for items with valid URNs.

**Updating show pages to use mps-embed for items with valid URNs.**
* * *

**JIRA Ticket**: [LTSVIEWER-250](https://at-harvard.atlassian.net/browse/LTSVIEWER-250)

# What does this Pull Request do?
This updates the item detail pages to use mps-embed to deliver the old Mirador viewer if a URN exists for the item. If there is not a valid URN, it falls back to the old iframe.

# How should this be tested?

A description of what steps someone could take to:
* Pull in the latest code from the `LTSVIEWER-250` branch
* Add the new `EMBED_URL` to your .env file. It is easiest to use QA for this: https://embed-qa.lib.harvard.edu
* Build CURIOSity off of the `LTSVIEWER-250` branch
* To test items without a URN, re-Harvest the Demo exhibit. An example item called "Shen lun ; Quan shan liang yan..." has been added to that exhibit today.
* Go to any item and confirm that the Mirador Viewer is still showing up. If you view the source you should see an <iframe> with width=100% and height=100% to confirm it is using mps-embed.
* Go to the "Shen lun ; Quan shan liang yan..."  item in the Demo exhibit. Confirm the Mirador Viewer is still showing up. f you view the source you should see an <iframe> with width=100% and height=600px to confirm it is not using mps-embed.

# Test coverage
Yes/No: Are changes in this pull-request covered by:
- unit tests? No
- integration tests? No

# Interested parties
@f8f8ff @enriquediaz
